### PR TITLE
final pull to stable

### PR DESCRIPTION
**Pull request by vanhauser-thc**
this is for the 3.12 release.

### CODE REVIEW

File: docs/env_variables.md [hunk 1]
```diff
@@ -419,7 +425,8 @@ checks or alter some of the more exotic semantics of the tool:
     no valid terminal was detected (for virtual consoles)
 
   - If you are Jakub, you may need `AFL_I_DONT_CARE_ABOUT_MISSING_CRASHES`.
-    Others need not apply.
+    Others need not apply, unless they also want to disable the
+    /proc/sys/kernel/core_pattern check.
```
Comment (domenukk) [inline docs/env_variables.md]: backticks here? `/proc/sys/kernel/core_pattern`

File: src/afl-common.c [hunk 2]
```diff
@@ -618,6 +618,102 @@ char *get_afl_env(char *env) {
...
 
+u8 extract_and_set_env(u8 *env_str) {
+
+  if (!env_str) { return 0; }
+
+  u8 *p = ck_strdup(env_str);
+
+  u8 *end = p + strlen((char *)p);
+
+  u8 ret_val = 0;  // return false by default
```
Comment (domenukk) [inline src/afl-common.c]: why not return `bool` here?

File: src/afl-fuzz.c [hunk 3]
```diff
@@ -1303,6 +1304,13 @@ int main(int argc, char **argv_orig, char **envp) {
 
   }
 
+  u8 *extra_env = (u8 *)getenv("AFL_TARGET_ENV");
```
Comment (domenukk) [inline src/afl-fuzz.c]: Does this not use the usual `afl->env` stuff on purpose?

File: src/afl-common.c [hunk 4]
```diff
@@ -618,6 +618,102 @@ char *get_afl_env(char *env) {
...
+
+  u8 *rest = p;
+  u8 *key = p;
+  u8 *val = p;
```
Comment (domenukk) [inline src/afl-common.c]: `key` and `val` are never read.

File: docs/Changelog.md [hunk 5]
```diff
@@ -27,6 +27,8 @@ sending a mail to <afl-users+subscribe@googlegroups.com>.
     - add AFL_QEMU_CUSTOM_BIN
   - unicorn_mode
     - accidently removed the subfolder from github, re-added
+  - added DEFAULT_PERMISSION to config.h for all files created, default
```
Comment (domenukk) [inline docs/Changelog.md:31]: `DEFAULT_PERMISSION` with backticks? Same for `0600`

**Review by domenukk — APPROVED** [review]

File: docs/Changelog.md (hunk 5 above)
Comment (vanhauser-thc) [reply docs/Changelog.md:31]: that would be then needed for most lines in the changelog ... nah :) rather spend my time otherwise :) but feel free!

**Pull request merged**